packages/components/molecules/f-mega-modal/src/components/MegaModal.vue

```diff
@@ -173,6 +191,14 @@ export default {
         },
         showAriaLabel () {
             return this.ariaLabel === '' ? this.uid : this.ariaLabel;
+        },
+        hasChevronIcon () {
+            return this.closeButtonStyle === 'chevron';
+        },
+        setCloseButtonIconStyle () {
+            return this.closeButtonStyle.includes('cross')
+                ? 'close-small-icon'
+                : 'chevron-left-icon';
```

Comment:
What do you think of this? Might make it easier to add other alternative icons in future?
```suggestion
            const iconMap = {
                chevron: 'chevron-left-icon',
                cross: 'close-small-icon'
            };
            return iconMap[this.closeButtonStyle];
```

Reply:
I did actually think about a similar approach (using switch tho) but left it as i think if we need more in future we could modify it then? I don't really seeing more than two icons ever being used in this situation to be honest.